Zilon.Core/Zilon.BotEnvironment/AutoplayEngine.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Extensions.DependencyInjection;
+
+using Zilon.Bot.Players;
+using Zilon.Bot.Sdk;
+using Zilon.Core.Scoring;
+using Zilon.Core.Tactics;
+using Zilon.Emulation.Common;
+
+namespace Zilon.BotEnvironment
+{
+    class AutoplayEngine : AutoplayEngineBase<IPluggableActorTaskSource>
+    {
+        private const string SCORE_FILE_PATH = "bot-scores";
+        private const int BOT_EXCEPTION_LIMIT = 3;
+        private const int ENVIRONMENT_EXCEPTION_LIMIT = 3;
+
+        private readonly Startup _startup;
+        private readonly string scoreFilePreffix;
+
+        int botExceptionCount = 0;
+        int envExceptionCount = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to 'envExceptionCount', the compiler will do that for you.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5028977)